analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocations.cs

```diff
@@ -136,42 +137,98 @@ End Sub
+        [DataRow("this.SomeExtensionOnObject();")]
+        [DataRow("Extensions.SomeExtensionOnObject(this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((IDisposable)this);")]
+        [DataRow("Extensions.SomeExtensionOnObject((object)(IDisposable)this);")]
+        [DataRow("((object)(IDisposable)this).SomeExtensionOnObject();")]
+        public void Invocation_InstanceMethodCallDoesClearFieldOnThis(string invocation)
+        {
+            var code = $@"
+using System;
+public class Sample: IDisposable
+{{
+    object field1;
+    object field2;
+    static object staticField1;
+    static object staticField2;
+
+    void Main()
+    {{
+        field1 = null;
+        staticField1 = null;
+        Tag(""BeforeField1"", field1);
+        Tag(""BeforeStaticField1"", staticField1);
+        {invocation}
+        Tag(""AfterField1"", field1);
+        Tag(""AfterStaticField1"", staticField1);
+    }}
+
+    private void Initialize() {{ }}
+    void IDisposable.Dispose() {{ }}
+    private static void Tag(string name, object arg) {{ }}
+}}
+
+public static class Extensions
+{{
+    public static void SomeExtensionOnSample(this Sample sample) {{ }}
+    public static void SomeExtensionOnObject(this object obj) {{ }}
+}}";
+            var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+            validator.ValidateContainsOperation(OperationKind.Invocation);
+            validator.ValidateTag("BeforeField1", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+            validator.ValidateTag("BeforeStaticField1", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+            validator.ValidateTag("AfterField1", x => x.Constraint<ObjectConstraint>().Should().BeNull());
+            validator.ValidateTag("AfterStaticField1", x => x.Constraint<ObjectConstraint>().Should().BeNull());
+        }
+
+        [DataTestMethod]
+        [DataRow("this?.InstanceMethod();")]
+        [DataRow("StaticMethod();")]
+        [DataRow("Sample.StaticMethod();")]
+        [DataRow("var dummy = Property;")]
+        [DataRow("var dummy = this.Property;")]
+        [DataRow("Property.InstanceMethod();")]
+        [DataRow("this.Property.InstanceMethod();")]
+        [DataRow("this.Property?.InstanceMethod();")]
+        public void Invocation_InstanceMethodCallDoesNotClearFieldForOtherAccess(string invocation)
         {
             var code = $@"
 public class Sample
 {{
-    object field1 = null;
-    object field2 = null;
-    static object staticField1 = null;
-    static object staticField2 = null;
+    object field1;
+    static object staticField1;
 
-    void CallToMethodsShouldResetFieldConstraints()
+    void Main()
     {{
-        field1 = new object();
-        field2 = new object();
-        staticField1 = new object();
-        staticField2 = new object();
+        field1 = null;
+        staticField1 = null;
+        Tag(""BeforeField1"", field1);
```

Comment:
From this line down, the `1` is not needed anymore
```suggestion
        Tag(""BeforeField"", field);
```